### PR TITLE
Optimized Google Translator 

### LINKED ISSUE #93: Optimizing Google Translator
### Description

Recently I tried fixing issues #90  #52 with PR #92  which was related to the google translator it was amazing to see the art of web-scraping the translated text using python but the following issues are needed to be solved for a better performance.
**************************************************************************************************************************
1. ```GOOGLE_CODES_TO_LANGUAGES``` dictionary defined in ```constants.py``` is not apt as per the translator link: https://translate.google.com/m

As per the translator database, we have a total of 108 languages with fixed abbreviations that are needed to be cross-checked again with dictionary. Following languages described in the dictionary have the following issues in them.
```
'zh': 'chinese'
'zh-cn': 'chinese (simplified)'
'zh-tw': 'chinese (traditional)'
```
As per the translator database,, we have just one version of the Chinese language whose abbreviation is defined by ```zh-CN```.
```
'tl': 'filipino'
'fil': 'Filipino'
```
As per the translator database, we have just one version of the Filipino language whose abbreviation is defined by ```tl```.
```
'iw': 'hebrew'
'he': 'Hebrew
```
As per the translator database, we have just one version of the Hebrew language whose abbreviation is defined by ```iw```.
```
Kinyarwanda, Odia, Tatar, Uyghur
```
As per the translator database, we have facility to translate them but these languages are not added to dictionary.
**************************************************************************************************************************
2. Why do we get this error for any of the following??  ```There is no support for the chosen language```
```
GoogleTranslator(source='de', target='myanmar').translate('Zwei minimale')
GoogleTranslator(source='de', target='English').translate('Zwei minimale')
GoogleTranslator(source='de', target='kurmanji').translate('Zwei minimale')
```
This is a very serious concern that is needed to be resolved as we are binding the user to enter the language in the specified format of our dictionary, I will explain all of them why it happens so.
The first problem arises because we need to describe the ```myanmar``` as ```myanmar (burmese)``` which forces the user to enter the additional name of language too in brackets.
The second problem arises because we need to describe the ```English``` as ```english``` that forces the user to follow case sensitivity as per the dictionary name.
The third problem arises because we need to describe the ```kurmanji``` as ```kurdish (kurmanji)``` which implies we can't use this translator with other names of any language (if any).
**************************************************************************************************************************

I would like to work on the above, so kindly assign me to fix these issues.

### PR DESCRIPTION
**Pull request by kuspia**
Close #93 

This PR optimizes the google translator by fixing the bugs and enhances the features as described below:

1. User can freely enter the name of any language without worrying about case sensitivity.
2. Addition of new languages like ```Kinyarwanda, Odia, Tatar, Uyghur```.
3. Multiple names language support feature, that is languages known by other names can be translated now.
 
As of now the following dictionary map the primary name of a language to its secondary names (other names). I have only mentioned the languages that were having other names on the [google translator](https://translate.google.com/m). However, this can be extended by adding more languages, and their multiple names if exist. The code is modified in a manner that extension will not lead to any further issues or errors.
```
GOOGLE_LANGUAGES_SECONDARY_NAMES = {
    'myanmar': ['burmese'],
    'odia': ['oriya'],
    'kurdish':  ['kurmanji']
}
```

To ensure the accuracy and correctness of these modifications addition of test_cases in ```test_google_trans.py``` has been done and passed, moreover, these test cases are well-designed to maintain the robustness to any future modifications.
*****************************************************************************************************************************

### CODE REVIEW

**Review by nothead31 — APPROVED** [review]
Approved changes. The tests seem to be robust and follow recommended testing practices. @nidhaloff should review as well and sign off before merging.

**Comment by nothead31** [comment]
@nidhaloff double check, but I think this is good to merge. Given the changes extend functionality and present more than a bugfix, I say this is sufficient for a minor update to 1.6.

**Comment by nothead31** [comment]
@kuspia review the documentation to see if anything needs updating to reflect your changes. Also add your name to the contributors list, create an appropriate changelog entry, and iterate the version number using `poetry version minor`. If you don't get to it before the merge, I can always make those changes after the fact.

**Comment by kuspia** [comment]
Please answer the following questions/suggestions 

1. I have imported the ```random``` module so do I need to specify it somewhere?
2. I was thinking to extend the dictionary named ```GOOGLE_LANGUAGES_SECONDARY_NAMES``` but for that I need some good relevant source if possible please suggest me.
3. I was planning to add a new dictionary in constants which contains the translation of a sentence from English to every language and in test file I was thinking to randomly test any five translations which will assure the accuracy to any future changes.
4. I hope the following features that I changed are also needed to be done for other translator so a issue can be created for the same so that other people can make the whole package more robust and efficient.

File: deep_translator/google_trans.py
Comment (kuspia) [reply deep_translator/google_trans.py:33]: It should be noted that we can't lower some of the abbreviations like ```zh-CN or zh-TW``` since in the translator database they are defined like this, therefore it is mandatory to check whether the source or target are not abbreviations and if the condition satisfies then we can lower the ```source``` and ```target```. I just concatenated source and target with ```_map``` to assign it as a new name.

I would also like to point that earlier in the code as shown  
```
 if self.is_language_supported(source, target):
            self._source, self._target = self._map_language_to_code(source.lower(), target.lower())
```

First-line was not passed with (lower) function which was causing the issue of case sensitivity for example ```English``` was discarded and raises the exception of language not supported.

PS: I assigned source and target with name 
```
source_lower
target_lower
```
for better clarity and readibilty.

File: deep_translator/constants.py
Comment (kuspia) [reply deep_translator/constants.py:124]: I would like to mention as per the translator database, ```Hebrew and Filpino``` are uniquely abbreviated as defined:
```
'tl': 'filipino'
'iw': 'hebrew'
```

Now I need to remove the above languages because they were redundant earlier both languages were already described above in the dictionary itself and moreover, as per the translator database we don't have any abbreviations like ```fil``` and ```he```
I would like to pin it that Filipino and filipino are not two different languages same is true for hebrew and Hebrew.

************************************
Yes, the languages that I added is supported by the translator website, you can check out for uyghur 
https://translate.google.com/m?sl=ug&tl=pt&hl=en

I have tallied each and every abbreviation with the languages so you need not worry.

Comment (nidhaloff) [reply deep_translator/constants.py:124]: ah alright, yes this is a good catch, thanks

File: deep_translator/google_trans.py
Comment (nidhaloff) [inline deep_translator/google_trans.py:74]: you can return False instead of 0. It is more readable

File: deep_translator/google_trans.py
Comment (nidhaloff) [inline deep_translator/google_trans.py:87]: `elif self.is_secondary(language) `would be sufficient in this case. You do not need to check more in python.

However, if you check this in` is_language_supported`, then you don't have to check this here again

File: deep_translator/google_trans.py
Comment (nidhaloff) [inline deep_translator/google_trans.py:101]: you are checking here if language is secondary.  So you don't have to check again in the map_to_languages function

**Review by nidhaloff — CHANGES_REQUESTED** [review]
I added some comments and request changes ;) and thanks for adding tests

File: deep_translator/constants.py
Comment (kuspia) [reply deep_translator/constants.py:140]: Should I move 
```
test_text_standard
TRANSLATED_RESULTS
```
 to test_google_trans.py file?

Actually, I randomly test any five cases that's what I am trying to do as we know there are around 108 languages so test file will run any 5 cases to validate it.

File: deep_translator/google_trans.py
Comment (kuspia) [reply deep_translator/google_trans.py]: If it is clear can I mark this as resolved?

File: deep_translator/google_trans.py
Comment (kuspia) [reply deep_translator/google_trans.py]: done

Comment (kuspia) [reply deep_translator/google_trans.py]: done

File: deep_translator/google_trans.py
Comment (kuspia) [reply deep_translator/google_trans.py]: done

File: deep_translator/google_trans.py
Comment (kuspia) [reply deep_translator/google_trans.py]: done

File: deep_translator/google_trans.py
Comment (kuspia) [reply deep_translator/google_trans.py]: done

File: deep_translator/google_trans.py
Comment (kuspia) [reply deep_translator/google_trans.py]: done

File: deep_translator/google_trans.py
Comment (nidhaloff) [reply deep_translator/google_trans.py]: @kuspia yes

File: deep_translator/constants.py
Comment (nidhaloff) [reply deep_translator/constants.py:140]: @kuspia creating a new file is better ;)

Comment (kuspia) [reply deep_translator/constants.py]: Hi 
I removed test data from constants.py 
tried placing test_data.py inside tests 
```
from test_data import test_text_standard, TRANSLATED_RESULTS
```

I met with the following error:
```
E   ModuleNotFoundError: No module named 'test_data'
```

This happened since test_data is not a part of igel package and I can't import this way.


Solution: Either I can put the test_data in test_google_trans file or do you have any better suggestions,I thought creating test_data would be nice since it may be used by other translators.

**Comment by nidhaloff** [comment]
@kuspia tests are failing because of test_data

**Comment by kuspia** [comment]
> @kuspia tests are failing because of test_data

Hi @nidhaloff they are not failing because of tests, you can see all of them passed now and the code is perfect in the above comitt.

The reason why they were failing is because in test_google_trans.py file I imported in this manner:
```
from test_data import test_text_standard, TRANSLATED_RESULTS
```

however, after commenting on it and importing the test_data content it is working.

**Review by nidhaloff — APPROVED** [review]

**Comment by kuspia** [comment]
@nidhaloff do any further changes are needed or edits from my side? If yes kindly mention it or let's close this PR by merging it.

**Pull request merged**
